zarr_etl_tools/utils/metadata.py

```diff
@@ -117,40 +119,135 @@ def static_metadata(cls):
         """
         ...
 
+    def check_stac_exists(self, title: str, stac_type: "StacType") -> bool:
+        """Check if a STAC entity exists in the backing store
+
+        Parameters
+        ----------
+        title : str
+            STAC Entity title
+        stac_type : StacType
+            Type of STAC entity (Catalog, Collection or Item)
+
+        Returns
+        -------
+        bool
+            Whether the entity exists in the backing store
+        """
+        if isinstance(self.store, IPLD):
+            return self.check_stac_on_ipns(title)
+        elif isinstance(self.store, Local):
+            return pathlib.Path.exists(get_local_stac_path(title, stac_type))
+        else:
+            raise NotImplementedError
+
+    def publish_stac(self, title: str, stac_content: dict, stac_type: "StacType"):
+        """Publish a STAC entity to the backing store
+
+        Parameters
+        ----------
+        title : str
+            STAC Entity title
+        stac_content : dict
+            content of the stac entity
+        stac_type : StacType
+            Type of STAC entity (Catalog, Collection or Item)
+        """
+        if isinstance(self.store, IPLD):
+            self.ipns_publish(title, self.ipfs_put(self.json_to_bytes(stac_content)))
+        elif isinstance(self.store, Local):
+            local_path = get_local_stac_path(title, stac_type)
+            local_path.parent.mkdir(exist_ok=True, parents=True)
+            with open(local_path, "w") as f:
+                json.dump(stac_content, f)
+        else:
+            raise NotImplementedError
+
+    def retrieve_stac(
```

Comment:
It's because it's not possible to determine whether a key is a collection or item from the name itself, but for local and s3 paths, it will determine where the STAC lives.